.github/workflows/build-and-test.yml

```diff
@@ -107,6 +107,10 @@ jobs:
         run: make misspell
       - name: checkdoc
         run: make checkdoc
+      - name: Check for go mod dependency changes
+        run: |
+          make gotidy
+          git diff --exit-code || (echo 'go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.' && exit 1)
```

Comment:
@codeboten @dmitryax did we not have this before?

Reply:
I'm as surprised as you are... it looks like gotidy was never added to the workflows in this repo when it was added to the contrib repo https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/d67d622569f7f4de6bfeb28ec0940c9e0b01b6e7